packages/next/server.d.ts

```diff
@@ -2,3 +2,5 @@ export { NextFetchEvent } from 'next/dist/server/web/spec-extension/fetch-event'
 export { NextRequest } from 'next/dist/server/web/spec-extension/request'
 export { NextResponse } from 'next/dist/server/web/spec-extension/response'
 export { NextMiddleware } from 'next/dist/server/web/types'
```

Comment:
I wonder why `NextMiddleware` is in the types, but not really exposed from the JS code

Reply:
I think that's necessary to type it properly when you write a Middleware using TypeScript.